cli/azd/pkg/project/service_config.go

```diff
@@ -5,46 +5,62 @@ import (
 
 	"github.com/azure/azure-dev/cli/azd/pkg/apphost"
 	"github.com/azure/azure-dev/cli/azd/pkg/ext"
-	"github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
 	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
 )
 
+const defaultComponentName = "main"
+
 type ServiceConfig struct {
-	// Reference to the parent project configuration
-	Project *ProjectConfig `yaml:"-"`
-	// The friendly name/key of the project from the azure.yaml file
-	Name string `yaml:"-"`
+	*ComponentConfig `yaml:",inline"`
```

Comment:
In practice one should no longer access these fields directly right?  Instead, you should pull them from the 0th' component in the `Components` slice?  These exist here just for marshalling purposes?

I'm wondering if the duplication is going to be confusing long term and we should figure out a way to separate the wire yaml format from the model type of ServiceConfig.